contracts/libraries/Assessment/AssessmentIncidentsLib.sol

```diff
@@ -0,0 +1,135 @@
+      uint deposit = expectedPayoutNXM * config.incidentAssessmentDepositPercentage / PERC_BASIS_POINTS;
+      nxm.transferFrom(msg.sender, address(this), deposit);
+    }
+  }
+
+  function saveIncident (
+    IAssessment.Incident calldata incident,
+    IAssessment.Incident[] storage incidents,
+    IAssessment.AffectedToken calldata affectedToken,
+    mapping(uint104 => IAssessment.AffectedToken) storage tokenAffectedByIncident
+  ) external {
+    uint104 nextId = uint104(incidents.length);
+    tokenAffectedByIncident[nextId] = affectedToken;
+    incidents.push(incident);
+  }
+
+  function redeemIncidentPayout (
+    mapping(uint => address payable) storage internalContracts,
+    IAssessment.Incident calldata incident,
+    uint32 coverId,
+    uint depeggedTokens
+  ) external {
+
+    require(
+      AssessmentVoteLib._getPollStatus(incident.poll) == IAssessment.PollStatus.ACCEPTED,
+      "The incident must be accepted"
+    );
+
+    require(block.timestamp >= incident.poll.end, "The incident is in cooldown period");
+
+    address payable coverOwner;
+    uint payoutAmount;
+    {
+      ICover coverContract = ICover(internalContracts[uint(IMasterAwareV2.ID.CO)]);
+      coverOwner = payable(coverContract.ownerOf(coverId));
+      (
+        uint24 productId,
+        /*uint8 payoutAsset*/,
+        /*uint8 deniedClaims*/,
+        uint96 amount,
+        uint32 start,
+        uint32 period
+      ) = coverContract.covers(coverId);
+
+      require (coverOwner == msg.sender, "Payout can only be redeemed by cover owner");
+      require(productId == incident.details.productId, "Incident id mismatch");
+      require(start <= incident.details.date, "Cover start date is after the incident");
+      require(start + period >= incident.details.date, "Cover end date is before the incident");
+      uint gracePeriod = 0; // [todo] Get from product
+      require(start + period + gracePeriod >= block.timestamp, "Grace period has expired");
+      // Should BURN_RATIO & DEDUCTIBLE_RATIO be stored in product details?
+      payoutAmount = depeggedTokens; // [todo] Calculate payout amount
+      require(payoutAmount <= amount, "Payout exceeds covered amount");
+      coverContract.performCoverBurn(coverId, coverOwner, payoutAmount);
+    }
+
+
+    // [todo] Replace payoutAddress with the member's address using the member id
+    address payable payoutAddress;
+    {
+      IMemberRoles memberRolesContract = IMemberRoles(internalContracts[uint(IMasterAwareV2.ID.MR)]);
+      payoutAddress = memberRolesContract.getClaimPayoutAddress(coverOwner);
+    }
+
+    {
+      IPool poolContract = IPool(internalContracts[uint(IMasterAwareV2.ID.P1)]);
+      address asset = poolContract.assets(incident.details.payoutAsset); // [todo]
+      bool succeeded = poolContract.sendClaimPayout(asset, payoutAddress, payoutAmount);
+      require(succeeded, "Incident payout failed");
+    }
+  }
```

Comment:
emit event for successful payout?

Reply:
must burn / edit the cover nft

Reply:
do we send the claimant the edited cover nft? or will they redeem it?

Reply:
All of that should happen in performCoverBurn not here. Ignore events for now.